### PR TITLE
Add circleci config and example test

### PR DESCRIPTION
**Pull request by matthewjablack**
This PR sets up Circle CI 

Circle CI will run `npm test` every time a new commit is added. https://circleci.com/gh/ConsenSys/chainabstractionlayer/2

The current setup uses image `mblackmblack/bitcoin-core-with-nvm:0.0.5`

https://github.com/ConsenSys/liquality_devops/pull/1
https://github.com/ConsenSys/liquality_devops/tree/master/bitcoin-core-with-npm

A mock test has been added under `test/test.js`. Once we have settled on an implementation of the architecture for chain abstraction layer proper tests can be added.

### CODE REVIEW

**Review by harshjv — COMMENTED** [review]
Would be better if we can introduce a simple workflow. For example, using `build` job to actual build library and `test` job to test it.

**Comment by harshjv** [comment]
Also, regarding the docker image with `nvm`, I would prefer just `bitcored` image and installation of `nvm` should be taken care by the test runner/CI in `setup` step.

Read up this: http://software.danielwatrous.com/a-review-of-docker/

**Comment by harshjv** [comment]
@mattBlackDesign looks good. can you open this PR to `master`?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - run:
+          name: test
+          command: npm test
+
+workflows:
+  version: 2
+  build_and_test:
+    jobs:
+      - build
+      - test
```
Comment (harshjv) [inline .circleci/config.yml]: How about?

```workflows:
  version: 2
  build_and_test:
    jobs:
      - test:
        requires: build
```

Comment (harshjv) [reply .circleci/config.yml]: and only keep `npm test` in `test` job?

Comment (matthewjablack) [reply .circleci/config.yml]: perfect

**Comment by harshjv** [comment]
Closing in favor of b8279012426787338fdeb051ae37013f77071131

**Pull request closed without merging**